src/System.Private.CoreLib/shared/System/MemoryExtensions.Fast.cs

```diff
@@ -141,6 +141,15 @@ public static int IndexOf(this ReadOnlySpan<char> span, ReadOnlySpan<char> value
                 return -1;
             }
 
+            if (comparisonType == StringComparison.Ordinal)
+            {
+                return SpanHelpers.IndexOf(
+                    ref MemoryMarshal.GetReference(span),
+                    span.Length,
+                    ref MemoryMarshal.GetReference(value),
+                    value.Length);
+            }
+
             if (GlobalizationMode.Invariant)
             {
                 return CompareInfo.InvariantIndexOf(span, value, string.GetCaseCompareOfComparisonCulture(comparisonType) != CompareOptions.None);
```

Comment:
If it is worth, we will never hit the `StringComparison.Ordinal` case below and the argument for `CompareInfo.Invariant.IndexOfOrdinal` can be `true`.

Reply:
changed